src/Campaigns/Actions/CacheCampaignData.php

```diff
@@ -0,0 +1,179 @@
+use Give\Campaigns\Models\Campaign;
+use Give\Donations\Models\Donation;
+
+/**
+ * @unreleased
+ *
+ * @uses givewp_donation_created hook
+ * @uses givewp_donation_updated hook
+ *
+ * Action used to update campaign's stats data
+ *
+ */
+class CacheCampaignData
+{
+    /**
+     * @unreleased
+     */
+    public function __invoke(int $donationId): void
+    {
+        $donation = Donation::find($donationId);
+
+        if (!$donation) {
+            return;
+        }
+
+        if ($donation->status->isComplete() || $donation->status->isRenewal()) {
+            as_enqueue_async_action('givewp_cache_campaign_data', [$donation->campaignId], 'givewp_campaigns_cache');;
+        }
+
+    }
+
+    /**
+     * Register AS action
+     *
+     * @unreleased
+     */
+    public function registerAction()
+    {
+        add_action('givewp_cache_campaign_data', function (int $campaignId) {
+            $this->handleCache($campaignId);
+        });
+    }
+
+    /**
+     * Handle campaign cache
+     * @unreleased
+     */
+    public function handleCache(int $campaignId): void
+    {
+        $campaign = Campaign::find($campaignId);
+
+        if (!$campaign) {
+            return;
+        }
+
+        $campaignsData = get_option('give_campaigns_data', []);
+        $campaignsSubscriptionData = get_option('give_campaigns_subscription_data', []);
+
+        // Prefill cache structure to ensure keys exist
+        $campaignsData = array_merge([
+            'amounts' => [],
+            'donationsCount' => [],
+            'donorsCount' => []
+        ], $campaignsData);
+
+        $campaignsSubscriptionData = array_merge([
+            'amounts' => [],
+            'donationsCount' => [],
+            'donorsCount' => []
+        ], $campaignsSubscriptionData);
+
+        $donations = CampaignsDataQuery::donations([$campaign->id]);
+        $subscriptions = CampaignsDataQuery::subscriptions([$campaign->id]);
+
+        $isCached = $isSubscriptionCached = false;
+
+        // Update cache only if it exists for this campaign
+        foreach ($campaignsData['amounts'] as $i => $data) {
+            if ($data['campaign_id'] == $campaign->id) {
+                $isCached = true;
+
+                $campaignsData['amounts'][$i] = $donations->collectIntendedAmounts()[0];
+                break;
+            }
+        }
+
+        foreach ($campaignsData['donationsCount'] as $i => $data) {
+            if ($data['campaign_id'] == $campaign->id) {
+                $campaignsData['donationsCount'][$i] = $donations->collectDonations()[0];
+                break;
+            }
+        }
+
+        foreach ($campaignsData['donorsCount'] as $i => $data) {
+            if ($data['campaign_id'] == $campaign->id) {
+                $campaignsData['donorsCount'][$i] = $donations->collectDonors()[0];
+                break;
+            }
+        }
+
+
+        // Update campaign subscriptions data
+        if (defined('GIVE_RECURRING_VERSION')) {
```

Comment:
We are querying the subscriptions a few lines above in this way: 

`$subscriptions = CampaignsDataQuery::subscriptions([$campaign->id]);`

But in this condition, we are preventing the update if the recurring add-on is not installed.

Thus, I think we need this same condition before querying the `$subscriptions` data so we can prevent call this query without a reason for that.